### PR TITLE
Introduce buy bulk products.

### LINKED ISSUE #360: Merge Buy, BuyMultiple

### PR DESCRIPTION
**Pull request by ipdae**
- based on #357 , #359 
- resolve #360

### CODE REVIEW

File: Lib9c/SerializeKeys.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        public const string LegacySetIdKey = "set_id";
+        public const string LegacySpineResourcePathKey = "spine_resource_path";
+
+        // Stat
+        public const string StatTypeKey = "stt";
+        public const string StatValueKey = "v";
+        public const string LegacyStatTypeKey = "statType";
+        public const string LegacyStatValueKey = "value";
+        public const string AdditionalStatValueKey = "av";
+        public const string LegacyAdditionalStatValueKey = "additionalValue";
```
Comment (area363) [inline Lib9c/SerializeKeys.cs:51]:
```suggestion
        public const string StatTypeKey = "stt";
        public const string StatValueKey = "v";
        public const string AdditionalStatValueKey = "av";
        public const string LegacyStatTypeKey = "statType";
        public const string LegacyStatValueKey = "value";
        public const string LegacyAdditionalStatValueKey = "additionalValue";
```

File: Lib9c/Action/SellCancellation.cs [hunk 2]
```diff
@@ -78,72 +85,82 @@ public override IAccountStateDelta Execute(IActionContext context)
...
+                if (!(rawShop is null))
+                {
+                    Dictionary legacyShopDict = (Dictionary) rawShop;
+                    Dictionary legacyProducts = (Dictionary) legacyShopDict[LegacyProductsKey];
+                    IKey productKey = (IKey) productId.Serialize();
+                    // SoldOut
+                    if (!legacyProducts.ContainsKey(productKey))
+                    {
+                        throw new ItemDoesNotExistException(
+                            $"{addressesHex}Aborted as the shop item ({productId}) was failed to get from the legacy shop."
```
Comment (area363) [inline Lib9c/Action/SellCancellation.cs:135]:
```suggestion
                            $"{addressesHex}Aborted as the shop item ({productId}) could not be found from the legacy shop."
```

File: Lib9c/Action/SellCancellation.cs [hunk 3]
```diff
@@ -78,72 +85,82 @@ public override IAccountStateDelta Execute(IActionContext context)
             var started = DateTimeOffset.UtcNow;
             Log.Verbose("{AddressesHex}Sell Cancel exec started", addressesHex);
 
-            if (!states.TryGetAgentAvatarStates(ctx.Signer, sellerAvatarAddress, out _, out var avatarState))
+            if (!states.TryGetAvatarState(ctx.Signer, sellerAvatarAddress, out var avatarState))
             {
-                return states;
+                throw new FailedLoadStateException(
+                    $"{addressesHex}Aborted as the avatar state of the seller was failed to load.");
```
Comment (area363) [inline Lib9c/Action/SellCancellation.cs:91]:
```suggestion
                    $"{addressesHex}Aborted as the avatar state of the seller failed to load.");
```

**Review by area363 — APPROVED** [review]

**Comment by longfin** [comment]
It seems to have conflicts

**Comment by ipdae** [comment]
@longfin resolve conflict.

**Review by longfin — APPROVED** [review]

**Pull request merged**